test/test-thread-pool.cpp

```diff
@@ -0,0 +1,775 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "test/test-thread-pool-util.hpp"
+#include "test/test-util.hpp"
+#include "thread-pool/thread-pool-executor.hpp"
+#include "thread-pool/thread-pool-vars-util.hpp"
+
+namespace intel {
+namespace hexl {
+
+#ifdef HEXL_MULTI_THREADING
+
+// Testing function that sets number of threads from env variable
+TEST(ThreadPool, setup_num_threads_env_var) {
+  // Max or default value result
+  auto max_or_default = std::min<uint64_t>(HEXL_DEFAULT_NUM_THREADS,
+                                           std::thread::hardware_concurrency());
+
+  // Overshooting: Max HW's value is set
+  {
+    char env[] = "HEXL_NUM_THREADS=999999999";
```

Comment:
string literal would be preferable here. It is not treated the same as char[].
```suggestion
    const char* env = "HEXL_NUM_THREADS=999999999";
```

Reply:
Const char is incompatible with **putenv** parameter char *. Then, if I don't mae it constant I get warnings: 

warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
   22 |     char* env = "HEXL_NUM_THREADS=999999999";

Reply:
ah. ok.